go/ocr2/decryptionplugin/decryption.go

```diff
@@ -248,12 +248,11 @@ func (dp *decryptionPlugin) Report(ctx context.Context, ts types.ReportTimestamp
 			}
 
 			validDecryptionShares[ciphertextID] = append(validDecryptionShares[ciphertextID], validDecryptionShare)
```

Comment:
I agree, we need to keep processing the observations.
What about the following suggestion? To make sure we don't aggregate more than f+1 shares. For efficiency.
```
	if len(validDecryptionShares[ciphertextID]) < fPlusOne {
		validDecryptionShares[ciphertextID] = append(validDecryptionShares[ciphertextID], validDecryptionShare)
	} else {
		dp.logger.Trace("DecryptionReporting Report: we have already f+1 valid decryption shares", commontypes.LogFields{
			"ciphertextID": ciphertextID,
			"observer":     ob.Observer,
		})
	}		
```

Reply:
Done.